packages/test/test-end-to-end-tests/src/test/refreshSerializerLifeCycle.spec.ts

```diff
@@ -0,0 +1,198 @@
+import { Deferred } from "@fluidframework/core-utils/internal";
+import {
+	ITestFluidObject,
+	timeoutPromise,
+	DataObjectFactoryType,
+	createAndAttachContainer,
+	timeoutAwait,
+	waitForContainerConnection,
+	type ChannelFactoryRegistry,
+	type ITestObjectProvider,
+} from "@fluidframework/test-utils/internal";
+import type { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import {
+	type IContainerRuntimeOptions,
+	DefaultSummaryConfiguration,
+} from "@fluidframework/container-runtime/internal";
+import type {
+	ConfigTypes,
+	IConfigProviderBase,
+	ITelemetryBaseLogger,
+} from "@fluidframework/core-interfaces/internal";
+import { SharedMap, type ISharedMap } from "@fluidframework/map/internal";
+import { MockLogger } from "@fluidframework/telemetry-utils/internal";
+import type { IHostLoader } from "@fluidframework/container-definitions/internal";
+import { wrapObjectAndOverride } from "../mocking.js";
+
+const testConfigs = generatePairwiseOptions({
+	savedOps: [true, false],
+	pendingOps: [true, false],
+	savedOps2: [true, false],
+	pendingOps2: [true, false],
+	summaryWhileOffline: [true, false],
+	waitForRefresh: [true,false],
+});
+
+describeCompat("Validate Attach lifecycle", "NoCompat", (getTestObjectProvider, apis) => {
+	const mapId = "map";
+	const registry: ChannelFactoryRegistry = [[mapId, SharedMap.getFactory()]];
+	const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+		getRawConfig: (name: string): ConfigTypes => settings[name],
+	});
+	const runtimeOptions: IContainerRuntimeOptions = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				...DefaultSummaryConfiguration,
+				...{
+					maxTime: 5000 * 12,
+					maxAckWaitTime: 120000,
+					maxOps: 1,
+					initialSummarizerDelayMs: 20,
+				},
+			},
+		},
+		enableRuntimeIdCompressor: "on",
+	};
+
+	const waitForSummary = async (container) => {
+		await timeoutPromise((resolve, reject) => {
+			let summarized = false;
+			container.on("op", (op: { type: string; }) => {
+				if (op.type === "summarize") {
+					summarized = true;
+				} else if (summarized && op.type === "summaryAck") {
+					resolve();
+				} else if (op.type === "summaryNack") {
+					reject(new Error("summaryNack"));
+				}
+			});
+		});
+	};
+
+	for (const testConfig of testConfigs) {
+		let provider: ITestObjectProvider;
+		let mockLogger: MockLogger;
+		let getLatestSnapshotInfoP: Deferred<void>;
+		let loader: IHostLoader;
+		let container: IContainerExperimental;
+		let url;
+		let map: ISharedMap;
+		let dataStore: ITestFluidObject;
+		let testContainerConfig;
+		beforeEach(async () => {
+			provider = getTestObjectProvider();
+			mockLogger = new MockLogger();
+			getLatestSnapshotInfoP = new Deferred<void>();
+			testContainerConfig = {
+				fluidDataObjectType: DataObjectFactoryType.Test,
+				registry,
+				runtimeOptions,
+				loaderProps: {
+					logger: wrapObjectAndOverride<ITelemetryBaseLogger>(mockLogger, {
```

Comment:
using a mock logger here will prevent logs from this test going into our telemetry system and being part of our test alerts, it also doesn't appear you leverage the mock logger specifically. there should be a logger available on the test provider that you can wrap instead